server/src/test/java/org/apache/kafka/server/MonitorablePluginsIntegrationTest.java

```diff
@@ -0,0 +1,111 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class MonitorablePluginsIntegrationTest {
+
+    private static int controllerId(Type type) {
+        return type == Type.KRAFT ? 3000 : 0;
+    }
+
+    private static Map<String, String> expectedTags(String config, String clazz) {
+        return expectedTags(config, clazz, Map.of());
+    }
+
+    private static Map<String, String> expectedTags(String config, String clazz, Map<String, String> extraTags) {
+        Map<String, String> tags = new LinkedHashMap<>();
+        tags.put("config", config);
+        tags.put("class", clazz);
+        tags.putAll(extraTags);
+        return tags;
+    }
+
+    @ClusterTest(
+        types = {Type.KRAFT, Type.CO_KRAFT},
+        serverProperties = {
+            @ClusterConfigProperty(key = StandardAuthorizer.SUPER_USERS_CONFIG, value = "User:ANONYMOUS"),
+            @ClusterConfigProperty(key = AUTHORIZER_CLASS_NAME_CONFIG, value = "org.apache.kafka.metadata.authorizer.StandardAuthorizer"),
+            @ClusterConfigProperty(key = REPLICA_SELECTOR_CLASS_CONFIG, value = "org.apache.kafka.server.MonitorablePluginsIntegrationTest$MonitorableReplicaSelector")
+        }
+    )
+    public void testMonitorableServerPlugins(ClusterInstance clusterInstance) {
+        assertAuthorizerMetrics(clusterInstance);
+        assertReplicaSelectorMetrics(clusterInstance);
+    }
+
+    private void assertAuthorizerMetrics(ClusterInstance clusterInstance) {
+        assertMetrics(
+                clusterInstance.brokers().get(0).metrics(),
+                4,
+                expectedTags(AUTHORIZER_CLASS_NAME_CONFIG, "StandardAuthorizer", Map.of("role", "broker")));
+
+        assertMetrics(
+                clusterInstance.controllers().get(controllerId(clusterInstance.type())).metrics(),
+                4,
+                expectedTags(AUTHORIZER_CLASS_NAME_CONFIG, "StandardAuthorizer", Map.of("role", "controller")));
+    }
+
+    private void assertReplicaSelectorMetrics(ClusterInstance clusterInstance) {
+        assertMetrics(
+                clusterInstance.brokers().get(0).metrics(),
+                MonitorableReplicaSelector.METRICS_COUNT,
+                expectedTags(REPLICA_SELECTOR_CLASS_CONFIG, MonitorableReplicaSelector.class.getSimpleName()));
+    }
+
+    private void assertMetrics(Metrics metrics, int expected, Map<String, String> expectedTags) {
+        int found = 0;
+        for (MetricName metricName : metrics.metrics().keySet()) {
+            if (metricName.group().equals("plugins")) {
+                Map<String, String> tags = metricName.tags();
+                if (expectedTags.equals(tags)) {
+                    found++;
+                }
+            }
+        }
+        assertEquals(expected, found);
+    }
+
+    public static class MonitorableReplicaSelector extends RackAwareReplicaSelector implements Monitorable {
+
+        private static final int METRICS_COUNT = 1;
```

Comment:
Excuse me, why to set a static constant in a inner class? Maybe it can be moved to `MonitorablePluginsIntegrationTest`?

Reply:
I've put it in the class as it's the number of metrics this class registers.